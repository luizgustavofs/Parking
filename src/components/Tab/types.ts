export interface TabProps {
  active?: boolean;
  label?: string;
  onClick: () => void;
}
