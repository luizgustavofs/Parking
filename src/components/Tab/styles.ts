import styled from 'styled-components';

import { TabProps } from './types';

export const Container = styled.div<TabProps>`
  display: flex;
  flex: 1;
  border-bottom: 3px solid ${({ theme }) => theme.colors.primary};
  border-top-right-radius: 4px;
  border-top-left-radius: 4px;
  align-items: center;
  justify-content: center;

  h3 {
    color: ${({ active, theme }) =>
      active ? theme.colors.primary : theme.colors.neutral._80};
  }

  background-color: ${({ active }) => (active ? '#fff' : 'transparent')};
  border-bottom-color: ${({ active, theme }) =>
    active ? theme.colors.primary : 'transparent'};
  border-bottom-width: 3px;
`;
