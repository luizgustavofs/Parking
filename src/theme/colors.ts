export const colors = {
  default: '#000000',
  primary: '#4DD0E1',
  secondary: '#F2F2F2',
  tertiary: '#28DD91',
  quaternary: '#A769B2',
  error: '#FF0068',
  baseColor: '#FFFBE6',
  neutral: {
    _00: '#ffffff',
    _70: '#DADADA',
    _80: '#9B9B9B',
  },
};
