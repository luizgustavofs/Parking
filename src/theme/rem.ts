export const REM_BASE = 16;
