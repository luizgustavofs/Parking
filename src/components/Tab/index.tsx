import React from 'react';

import { Container } from './styles';
import { TabProps } from './types';

const Tab: React.FC<TabProps> = ({ label, active, onClick }) => {
  return (
    <Container active={active} onClick={onClick}>
      <h3>{label}</h3>
    </Container>
  );
};

export default Tab;
