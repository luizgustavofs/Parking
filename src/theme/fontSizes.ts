import { REM_BASE } from './rem';

export const fontSizes = {
  h1: 38 / REM_BASE,
  h2: 24 / REM_BASE,
  h3: 16 / REM_BASE,
  h4: 15 / REM_BASE,
  h5: 12 / REM_BASE,
};
