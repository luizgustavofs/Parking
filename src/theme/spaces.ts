import { REM_BASE } from './rem';

export const spaces = {
  x1: 4 / REM_BASE,
  x2: 8 / REM_BASE,
  x3: 12 / REM_BASE,
  x4: 16 / REM_BASE,
  x6: 24 / REM_BASE,
  x8: 32 / REM_BASE,
  x10: 40 / REM_BASE,
  x11: 44 / REM_BASE,
  x12: 48 / REM_BASE,
  x13: 52 / REM_BASE,
  x14: 56 / REM_BASE,
  x16: 64 / REM_BASE,
  x18: 72 / REM_BASE,
  x20: 80 / REM_BASE,
  x22: 88 / REM_BASE,
  x24: 96 / REM_BASE,
  x30: 120 / REM_BASE,
  x40: 160 / REM_BASE,
  x50: 200 / REM_BASE,
};
