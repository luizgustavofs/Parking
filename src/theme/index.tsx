import { ReactNode } from 'react';

import { ThemeProvider } from 'styled-components';

import { colors } from './colors';
import { fontSizes } from './fontSizes';
import { spaces } from './spaces';

export const theme = {
  colors,
  spaces,
  fontSizes,
};

const ProjectThemeProvider: React.FC<{ children: ReactNode }> = ({
  children,
  ...props
}) => {
  return (
    <ThemeProvider theme={theme} {...props}>
      {children}
    </ThemeProvider>
  );
};

export { ProjectThemeProvider };
