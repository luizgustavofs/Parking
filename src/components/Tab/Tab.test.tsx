import { create } from 'react-test-renderer';
import { ThemeProvider } from 'styled-components';

import Tab from '.';
import { theme } from '../../theme';

describe('Tab', () => {
  it('should render correctly with variants', () => {
    const { toJSON } = create(
      <ThemeProvider theme={theme}>
        <Tab onClick={jest.fn} />
      </ThemeProvider>,
    );
    expect(toJSON()).toMatchSnapshot();
  });
});
