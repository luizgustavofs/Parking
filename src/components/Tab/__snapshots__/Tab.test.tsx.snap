// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tab should render correctly with variants 1`] = `
<div
  className="sc-bczRLJ brxjCC"
  onClick={[Function]}
>
  <h3 />
</div>
`;
